src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -3305,67 +3455,85 @@ public static void printSiteInstInfo(SiteInst siteInst, PrintStream ps) {
+                        } else if (srcNetAlias.isVCC()) {
+                            parentPhysNet = design.getVccNet();
+                        }
 
-                if (!hierNet.equals(parentHierNet)) {
-                    String parentNetName = parentHierNet.getNet().getName();
-                    // Assume that a net named <const1> or <const0> is always a VCC or GND net
-                    if (parentNetName.equals(EDIFTools.LOGICAL_VCC_NET_NAME)) {
-                        parentPhysNet = design.getVccNet();
-                    } else if (parentNetName.equals(EDIFTools.LOGICAL_GND_NET_NAME)) {
-                        parentPhysNet = design.getGndNet();
-                    } else {
-                        parentPhysNet = design.getNet(parentHierNet.getHierarchicalNetName());
+                        if (!hierNet.equals(parentHierNet)) {
+                            String parentNetName = parentHierNet.getNet().getName();
+                            // Assume that a net named <const1> or <const0> is always a VCC or GND net
+                            if (parentNetName.equals(EDIFTools.LOGICAL_VCC_NET_NAME)) {
+                                parentPhysNet = design.getVccNet();
+                            } else if (parentNetName.equals(EDIFTools.LOGICAL_GND_NET_NAME)) {
+                                parentPhysNet = design.getGndNet();
+                            } else {
+                                parentPhysNet = design.getNet(parentHierNet.getHierarchicalNetName());
+                            }
+
+                            if (parentPhysNet == null) {
+                                synchronized (design) {
+                                    if (!net.rename(parentHierNet.getHierarchicalNetName())) {
+                                        System.out.println("WARNING: Failed to adjust physical net name " + net.getName());
+                                    }
+                                }
+                            }
+                        }
                     }
 
                     if (parentPhysNet != null) {
-                        // Fall through
-                    } else if (net.rename(parentHierNet.getHierarchicalNetName())) {
-                        // Fall through
-                    } else {
-                        System.out.println("WARNING: Failed to adjust physical net name " + net.getName());
+                        synchronized (design) {
+                            design.movePinsToNewNetDeleteOldNet(net, parentPhysNet, true);
+                        }
                     }
                 }
+            });
+            if (f != null) {
+                futures.add((Future<Void>) f);
             }
-
-            if (parentPhysNet != null) {
-                design.movePinsToNewNetDeleteOldNet(net, parentPhysNet, true);
-            }
+        }
+        while (!futures.isEmpty()) {
+            ParallelismTools.joinFirst(futures);
         }
```

Comment:
`joinFirst()` is for when you want to get the result from the first Future that completes so that it can be operated on. You want this instead:

```suggestion
        ParallelismTools.join(futures);
```